dataactvalidator/config/sqlrules/d41_detached_award_financial_assistance_3.sql

```diff
@@ -11,26 +11,13 @@ SELECT
     dafa.place_of_performance_code,
     dafa.place_of_performance_zip4a
 FROM detached_award_financial_assistance_d41_3_{0} AS dafa
-WHERE CASE WHEN COALESCE(dafa.place_of_performance_zip4a, '') != ''
-        -- 5 digit zip
-        THEN (CASE WHEN dafa.place_of_performance_zip4a ~ '^\d\d\d\d\d$'
-             THEN dafa.row_number NOT IN (SELECT DISTINCT sub_dafa.row_number
-                              FROM detached_award_financial_assistance_d41_3_{0} AS sub_dafa
-                              JOIN zips
-                                ON UPPER(LEFT(sub_dafa.place_of_performance_code, 2)) = zips.state_abbreviation
-                                    AND sub_dafa.place_of_performance_zip4a = zips.zip5)
-             -- 9 digit zip
-             ELSE (CASE WHEN dafa.place_of_performance_zip4a ~ '^\d\d\d\d\d\-?\d\d\d\d$'
-                    THEN dafa.row_number NOT IN (SELECT DISTINCT sub_dafa.row_number
-                                    FROM detached_award_financial_assistance_d41_3_{0} AS sub_dafa
-                                    JOIN zips
-                                      ON UPPER(LEFT(sub_dafa.place_of_performance_code, 2)) = zips.state_abbreviation
-                                        AND LEFT(sub_dafa.place_of_performance_zip4a, 5) = zips.zip5
-                                        AND SUBSTRING(REPLACE(sub_dafa.place_of_performance_zip4a, '-', ''), 6, 4) = zips.zip_last4)
-                    -- if any other format, pass (will check format in another rule)
-                    ELSE FALSE
-                    END)
-             END)
-        -- if not zip4, just let it pass
-        ELSE FALSE
-        END
\ No newline at end of file
+WHERE CASE WHEN (COALESCE(dafa.place_of_performance_zip4a, '') != ''
+                 AND (dafa.place_of_performance_zip4a ~ '^\d\d\d\d\d$'
```

Comment:
We still need 2 WHEN clauses here because the comparison for 5 digits vs 9 digits is different (what if someone enters `12345` and we compare 5-digit `12345` and 4-digit `2345` and there's no 4-digit `2345` but the 5-digit (which is all they entered) is still valid? This isn't being checked for in this rule right now.) If we're ONLY checking 5-digit here (which it seems like the intent is), then the wording is incorrect in the rule description.